drivers/bluetooth/hci/userchan.c

```diff
@@ -289,9 +329,29 @@ static void add_btuserchan_arg(void)
 
 static void btuserchan_check_arg(void)
 {
-	if (bt_dev_index < 0) {
-		posix_print_error_and_exit("Error: Bluetooth device missing. "
-					   "Specify one using --bt-dev=hciN\n");
+	if (!arg_found) {
+		posix_print_error_and_exit("Error: Bluetooth device missing.\n"
+					   "Specify either a local hci interface --bt-dev=hciN\n"
+					   "or a valid hci tcp server --bt-dev=ip_address:port\n");
+	}
+	if (hci_socket) {
+		if (arg_hci_idx > 0 && arg_hci_idx <= USHRT_MAX) {
+			bt_dev_index = arg_hci_idx;
```

Comment:
Since `bt_dev_index` is global, I suggest to make `arg_hci_idx` local, and do this check where `arg_hci_idx` is assigned. IMO there should be no reason to keep `arg_hci_idx` as a global value

Reply:
I made `arg_hci_idx` global to leave it for discussion. The question for me is where should  user argument values be checked?

Directly when `cmd_bt_dev_found` is called or `btuserchan_check_arg` ? 

If the bt index would be checked in `cmd_bt_dev_found` would it not make sense to also check the other arguments (i.e., ip address, port number) in this function ? 

It was not clear to me if `btuserchan_check_arg` was just used to check if the argument `--bt-dev` was set or if it should also check all other arguments.

Reply:
> Directly when cmd_bt_dev_found is called or btuserchan_check_arg ?

It is a bit better to check directly in `cmd_bt_dev_found`, no need to keep going if there is a bad cmd line argument.

> would it not make sense to also check the other arguments (i.e., ip address, port number) in this function ?

Yep